internal/cli/project_inspect.go

```diff
@@ -0,0 +1,261 @@
+	"strings"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type ProjectInspectCommand struct {
+	*baseCommand
+
+	flagJson bool
+}
+
+func (c *ProjectInspectCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithConfig(true), // optional config loading
+	); err != nil {
+		return 1
+	}
+
+	cmdArgs := flagSet.Args()
+	var projectTarget string
+	if len(cmdArgs) > 1 {
+		c.ui.Output("No more than 1 argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	} else if len(cmdArgs) == 0 {
+		// If we're in a project dir, load the name. Otherwise we'll
+		// try the arg passed in
+		if c.project.Ref() != nil {
+			projectTarget = c.project.Ref().Project
+		} else {
+			c.ui.Output("Project argument required, and not in a project directory..\n\n"+
+				c.Help(), terminal.WithErrorStyle())
+			return 1
+		}
+	} else if len(cmdArgs) == 1 {
+		// project requested
+		projectTarget = cmdArgs[0]
+	}
+
+	err := c.FormatProject(projectTarget)
+	if err != nil {
+		c.ui.Output("Failed to format project: %s"+
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle())
+		return 1
+	}
+	return 0
+}
+
+func (c *ProjectInspectCommand) FormatProject(projectTarget string) error {
+	// Get our API client
+	client := c.project.Client()
+
+	resp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+		Project: &pb.Ref_Project{
+			Project: projectTarget,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	project := resp.Project
+	workspaces := resp.Workspaces
+
+	var appNames []string
+	for _, app := range project.Applications {
+		appNames = append(appNames, app.Name)
+	}
+
+	var workspaceNames []string
+	for _, ws := range workspaces {
+		workspaceNames = append(workspaceNames, ws.Workspace.Workspace)
+	}
+
+	dataSource := "Local" // if unset, assume local
+	if project.DataSource != nil {
+		switch project.DataSource.Source.(type) {
+		case *pb.Job_DataSource_Local:
+			dataSource = "Local"
+		case *pb.Job_DataSource_Git:
+			dataSource = "Git"
+		}
+	}
+
+	var projectPollEnabled bool
+	var projectPollInterval string
+	if project.DataSourcePoll != nil {
+		projectPollEnabled = project.DataSourcePoll.Enabled
+		projectPollInterval = project.DataSourcePoll.Interval
+	}
+
+	var appPollEnabled bool
+	var appPollInterval string
+	if project.StatusReportPoll != nil {
+		appPollEnabled = project.StatusReportPoll.Enabled
+		appPollInterval = project.StatusReportPoll.Interval
+	}
+
+	if c.flagJson {
+		projectHeaders := []string{
+			"Project", "Applications", "Workspaces", "Remote Enabled", "Data Source", "Project Poll Enabled",
+			"Project Poll Interval", "App Status Poll Enabled", "App Status Poll Interval",
```

Comment:
Can we show everything that can be configured with `waypoint project apply`? Looks like we're missing `-git-url` and `-git-path`, among other things.

Reply:
Yes. If those values are unset they won't show up in a NamedValue output though.

Reply:
Gotcha. 

I think it's confusing that the `-poll-interval` flag in `project apply`, the 
[`data_source_poll { enabled }` proto](https://github.com/hashicorp/waypoint/blob/main/internal/server/proto/server.proto#L596), and `Project Poll Enabled` here all refer to the same thing. I feel like "Data source polling" is a good vernacular to standardize around, because that's what we're actually polling?

Looking at the project proto, it looks like we're still missing:
- `file_change_signal` (also missing from `waypoint project apply`)
- `waypoint_hcl_format`

Reply:
I can show the file change signal, but I'd probably omit the waypoint hcl format, I'm not sure how useful that is or what you'd do with that information 🤔 Also got the headers replaced with _Data Source Poll_!